scripts/install.sh

```diff
@@ -63,54 +65,50 @@ shift $(expr $OPTIND - 1) # remove options from positional parameters
 # Overwrite default properties
 CONFIG_DIR_PATH=${config_dir_path:-$CONFIG_DIR_PATH_DEFAULT}
 NETWORK=${network:-$NETWORK_DEFAULT}
-CONTRACTS_ONLY=${contracts_only:-false}
+REUSE_CONTRACTS=${reuse_contracts:-false}
+
+echo "reuse contracts::" ${REUSE_CONTRACTS}
 
 # Run script
 printf "${LOG_START}Starting installation...${LOG_END}"
 
 printf "Config dir path: $CONFIG_DIR_PATH\n"
 printf "Network: $NETWORK"
 
-cd $KEEP_CORE_SOL_PATH
+cd $KEEP_BEACON_SOL_PATH
 
-printf "${LOG_START}Installing NPM dependencies...${LOG_END}"
-npm install
+printf "${LOG_START}Installing YARN dependencies...${LOG_END}"
+# rm -rf node_modules/
+yarn install
 
-if [ "$NETWORK" == "local" ]; then
+if [ "$NETWORK" == "development" ]; then
     printf "${LOG_START}Unlocking ethereum accounts...${LOG_END}"
     KEEP_ETHEREUM_PASSWORD=$KEEP_ETHEREUM_PASSWORD \
-        npx truffle exec scripts/unlock-eth-accounts.js --network $NETWORK
+        npx hardhat run scripts/unlock-eth-accounts.ts --network $NETWORK
 fi
 
-printf "${LOG_START}Migrating contracts...${LOG_END}"
-rm -rf build/
+printf "${LOG_START}Building contracts...${LOG_END}"
+# rm -rf build && rm -rf cache && rm -rf typechain
 
-CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
-    npx truffle migrate --reset --network $NETWORK
+yarn build
 
-printf "${LOG_START}Copying contract artifacts...${LOG_END}"
-rm -rf artifacts
-cp -r build/contracts artifacts
-npm link
+if [ "$REUSE_CONTRACTS" = false ] ; then
+   printf "${LOG_START}Deploying contracts...${LOG_END}"
+   rm -rf deployments/development/
 
+   CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
+      npx hardhat deploy --reset --export export.json --network $NETWORK
+fi
+    
 printf "${LOG_START}Initializing contracts...${LOG_END}"
 
 CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
-    npx truffle exec scripts/delegate-tokens.js --network $NETWORK
-
-if [ "$CONTRACTS_ONLY" = false ] ; then
-  printf "${LOG_START}Updating keep-core client configs...${LOG_END}"
-  for CONFIG_FILE in $CONFIG_DIR_PATH/*.toml
-  do
-      KEEP_CORE_CONFIG_FILE_PATH=$CONFIG_FILE \
-      CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
-          npx truffle exec scripts/lcl-client-config.js --network $NETWORK
-  done
-
-  printf "${LOG_START}Building keep-core client...${LOG_END}"
-  cd $KEEP_CORE_PATH
-  go generate ./...
-  go build -a -o keep-core .
-fi
+    npx hardhat run scripts/init-contracts.ts --network $NETWORK
+
+printf "${LOG_START}Building keep-core client...${LOG_END}"
```

Comment:
To maintain the previous functionality of `CONTRACTS_ONLY` flag, maybe we could introduce `--skip-client-build` flag? This would be consistent with `--skip-deployment` flag.

Reply:
Sure, we can do this.

Part of: [2dcb842](https://github.com/keep-network/keep-core/pull/3026/commits/2dcb842ce661077ccb2491e425aa08f4112bbdbe)